components/equihash/src/params.rs

```diff
@@ -0,0 +1,37 @@
+#[derive(Clone, Copy)]
+pub(crate) struct Params {
+    pub(crate) n: u32,
+    pub(crate) k: u32,
+}
+
+impl Params {
+    /// Returns `None` if the parameters are invalid.
```

Comment:
Optional/Nitpick: More verbose docs, maybe with a reference to where and/or how the parameters are used? 

```suggestion
    /// Accepts `n` and `k` values to use as Equihash parameters.
    ///
    /// Returns a new instance of [`Params`] if the provided values are valid, or
    /// `None` if the parameters are invalid.
```

Reply:
The doc change here was just to clarify the change from `Error` (there is only one "error" case here, and `Error` is about verification errors). I've opened #1358 for the general case of documenting the crate-private methods.